third-party/ios/ExecutorchLib/ExecutorchLib/Utils.mm

```diff
@@ -62,42 +76,58 @@
+
   if (result.ok()) {
-    Tensor outputTensor = result->at(0).toTensor();
-    numel = outputTensor.numel();
-    return outputTensor.const_data_ptr<T>();
+    std::vector<std::span<const T>> outputVec;
+
+    for (const auto &currentResult : *result) {
+      Tensor currentTensor = currentResult.toTensor();
+      std::span<const T> currentSpan(currentTensor.const_data_ptr<T>(), currentTensor.numel());
+      outputVec.push_back(std::move(currentSpan));
+    }
+    return outputVec;
   }
+
   @throw [NSException
       exceptionWithName:@"forward_error"
                  reason:[NSString stringWithFormat:@"%d", (int)result.error()]
                userInfo:nil];
 }
 
-template const int8_t *runForwardFromNSArray<int8_t>(
-    NSArray *inputArray, ssize_t &numel, std::vector<int> shapes,
+template std::vector<std::span<const int8_t>> runForwardFromNSArray<int8_t>(
+    NSArray *inputArray, std::vector<int> shapes,
     std::unique_ptr<executorch::extension::Module> &model);
-template const int32_t *runForwardFromNSArray<int32_t>(
-    NSArray *inputArray, ssize_t &numel, std::vector<int> shapes,
+template std::vector<std::span<const int32_t>> runForwardFromNSArray<int32_t>(
+    NSArray *inputArray, std::vector<int> shapes,
     std::unique_ptr<executorch::extension::Module> &model);
-template const int64_t *runForwardFromNSArray<int64_t>(
-    NSArray *inputArray, ssize_t &numel, std::vector<int> shapes,
+template std::vector<std::span<const int64_t>> runForwardFromNSArray<int64_t>(
+    NSArray *inputArray, std::vector<int> shapes,
     std::unique_ptr<executorch::extension::Module> &model);
-template const float *runForwardFromNSArray<float>(
-    NSArray *inputArray, ssize_t &numel, std::vector<int> shapes,
+template std::vector<std::span<const float>> runForwardFromNSArray<float>(
+    NSArray *inputArray, std::vector<int> shapes,
     std::unique_ptr<executorch::extension::Module> &model);
-
-template const double *runForwardFromNSArray<double>(
-    NSArray *inputArray, ssize_t &numel, std::vector<int> shapes,
+template std::vector<std::span<const double>> runForwardFromNSArray<double>(
+    NSArray *inputArray, std::vector<int> shapes,
     std::unique_ptr<executorch::extension::Module> &model);
 
+template NSArray *
+arrayToNSArray<int8_t>(const std::vector<std::span<const int8_t>> &dataPtrVec);
+template NSArray *arrayToNSArray<int32_t>(
+    const std::vector<std::span<const int32_t>> &dataPtrVec);
+template NSArray *arrayToNSArray<int64_t>(
+    const std::vector<std::span<const int64_t>> &dataPtrVec);
+template NSArray *
+arrayToNSArray<float>(const std::vector<std::span<const float>> &dataPtrVec);
+template NSArray *
+arrayToNSArray<double>(const std::vector<std::span <const double>> &dataPtrVec);
+
 template NSArray *arrayToNSArray<int8_t>(const void *array, ssize_t numel);
 template NSArray *arrayToNSArray<int32_t>(const void *array, ssize_t numel);
 template NSArray *arrayToNSArray<int64_t>(const void *array, ssize_t numel);
```

Comment:
this is getting a bit out of hand, let's just move all this code to the header file